app/assets/src/components/views/bulk_download/BulkDownloadModal.jsx

```diff
@@ -1,48 +1,18 @@
 import React from "react";
 import PropTypes from "prop-types";
-import { unset, find, get, set } from "lodash/fp";
-import memoize from "memoize-one";
 
+import {
+  getBackgrounds,
+  uploadedByCurrentUser,
+  getHeatmapMetrics,
+} from "~/api";
 import {
   getBulkDownloadTypes,
   validateSampleIdsForBulkDownload,
 } from "~/api/bulk_downloads";
 import Modal from "~ui/containers/Modal";
 
-import ChooseStep from "./ChooseStep";
-import ReviewStep from "./ReviewStep";
-
-const assembleSelectedDownload = memoize(
```

Comment:
I do not have that much context, so it is harder for me to grasp some of the changes and this could be a naive question. 
But what is the reasoning to move this function to the Choose step / Chooser?

Sees to me that the chooser object should be just a way to select options. But the main class would handle the actual download... no?

Reply:
I thought it would make for clearer code if all the async requests were done in the `BulkDownloadModal` component and then the resulting data passed down as props to the `ChooseStep`. The purpose of the callback functions I moved was to take data from the `ChooseStep`, assemble a download and pass it to the `ReviewStep`. Since we're eliminating the `ReviewStep` I thought it would be better to put all the logic for handling the selected options and creating a download in the `ChooseStep`.

I agree that I should name it something clearer, I'll do that.